### PR TITLE
Bug 1829600 - Create new "No internet connection" and "Address not found" UI tests

### PR DESCRIPTION
**Pull request by AndiAJ**
Bug 1829600 - Create new "No internet connection" and "Address not found" UI tests

✅ Both tests successfully passed 100x on Firebase.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1829600

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt [hunk 1]
```diff
@@ -112,4 +112,54 @@ class BrowsingErrorPagesTest {
...
+
+    @Test
+    fun noInternetConnectionErrorMessageTest() {
+        val firstWebsite = "https://mozilla-mobile.github.io/testapp/v2.0/loginForm.html"
+        val secondWebsite = "www.example.com"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(firstWebsite.toUri()) {
+            waitForPageToLoad()
+            verifyPageContent("Login Form")
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt:144]: This part doesn't seem relevant for the test.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt [hunk 2]
```diff
@@ -112,4 +112,54 @@ class BrowsingErrorPagesTest {
...
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(invalidUrl.toUri()) {
+            waitForPageToLoad()
+            verifyAddressNotFoundErrorMessage()
+            clickTryAgainErrorPageButton()
+            verifyAddressNotFoundErrorMessage()
+        }
+
+        navigationToolbar {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt:129]: Is this part relevant to the test?

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt [hunk 3]
```diff
@@ -112,4 +112,54 @@ class BrowsingErrorPagesTest {
...
+        }.openNewTab {
+        }.submitQuery(secondWebsite) {
+            waitForPageToLoad()
+            verifyNoInternetConnectionErrorMessage()
+        }
+
+        setNetworkEnabled(true)
+
+        browserScreen {
+            clickTryAgainErrorPageButton()
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt:160]: Could you please replace this with clickPageObject(itemWithText("errorTryAgain"))?

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 4]
```diff
@@ -1126,6 +1126,28 @@ class BrowserRobot {
...
+            itemContainingText("The browser could not find the host server for the provided address."),
+            itemContainingText("Check the address for typing errors such as "),
+            itemContainingText("ww"),
+            itemContainingText(".example.com instead of "),
+            itemContainingText("www"),
+            itemContainingText(".example.com."),
+            itemContainingText("If you are unable to load any pages, check your device’s data or Wi-Fi connection."),
+        )
+        assertItemWithResIdExists(itemWithResId("errorTryAgain"))
+    }
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:1141]: I really don't know how stringent we need to be to checking for string precision here, this feels a bit excessive. I think it would be fine if we just check for the presence of a view ID or header to confirm that in view. What do you think @sv-ohorvath @AndiAJ

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt (hunk 3 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt]: Done !

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt (hunk 2 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt]: As discussed, I've removed it.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt [hunk 5]
```diff
@@ -112,4 +112,54 @@ class BrowsingErrorPagesTest {
             verifyPageContent("Example Domain")
         }
     }
+
+    @Test
+    fun addressNotFoundErrorMessageTest() {
+        val invalidUrl = "ww.example.com"
+        val validUrl = "www.example.com"
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BrowsingErrorPagesTest.kt]: I've tried, but it doesn't work with localhost.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt (hunk 4 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:1132]: Sounds good @AaronMT

**Comment by AndiAJ** [comment]
@sv-ohorvath removed the unnecessary steps.
@AaronMT made the changes as suggested (left only the header of the error pages and the reload/try again button)

Thanks for the reviews ! ☺️ 
Please share your thoughts.
